SimCalorimetry/HcalTrigPrimProducers/plugins/HcalTrigPrimDigiProducer.cc

```diff
@@ -177,15 +184,26 @@ void HcalTrigPrimDigiProducer::beginRun(const edm::Run& run, const edm::EventSet
     // Filter weight represented in fixed point 8 bit
     int fixedPointWeight = -1;
 
+    // Filter veto threshold in fixed point 11 bit
+    int vetoThreshold = -1;
+
     // The absence of TT channels in the HcalTPChannelParameters
     // is intepreted as to not use the new filter
     auto tpParam = db->getHcalTPChannelParameter(hcalTTDetId, false);
-    if (tpParam)
+    if (tpParam) {
       fixedPointWeight = tpParam->getauxi1();
+      vetoThreshold = tpParam->getauxi2();
+    }
 
     // Do not let ieta 29 in the map
     // If the aieta already has a weight in the map, then move on
     if (aieta <= lastHBRing) {
+      if (!overrideDBvetoThresholdsHB_) {
```

Comment:
I think that this whole for loop, from [L177](https://github.com/cms-sw/cmssw/pull/44373/files#diff-0ba33b63e215e5b804454f70160398f6e8b39cb1ca188b08076bfbc88aaf20e2R177) until [L236](https://github.com/cms-sw/cmssw/pull/44373/files#diff-0ba33b63e215e5b804454f70160398f6e8b39cb1ca188b08076bfbc88aaf20e2R236) could be simplified as:

```C++
  for (std::vector<HcalElectronicsId>::const_iterator eId = vIds.begin(); eId != vIds.end(); eId++) {
    HcalTrigTowerDetId hcalTTDetId(emap->lookupTrigger(*eId));
    if (hcalTTDetId.null())
      continue;

    int aieta = abs(hcalTTDetId.ieta());
    // Do not let ieta 29 in the map
    if (aieta >= lastHERing)
      continue

    // Filter weight represented in fixed point 8 bit
    int fixedPointWeight = 255;
    // Filter veto threshold in fixed point 11 bit
    int vetoThreshold = 2047;
    // Number of filter presamples
    int presamples = 0;

    // The absence of TT channels in the HcalTPChannelParameters
    // is intepreted as to not use the new filter
    auto tpParam = db->getHcalTPChannelParameter(hcalTTDetId, false);
    if (tpParam) {
      // Fix number of filter presamples to one if we are using DB weights
      // Size of filter is already known when using DB weights
      // Weight from DB represented as 8-bit integer
      fixedPointWeight = tpParam->getauxi1();
      vetoThreshold = tpParam->getauxi2();
      presamples = 1;
    }

    // If the aieta already has a weight in the map, then move on
    if (aieta <= lastHBRing) {
      if (!overrideDBvetoThresholdsHB_) {
        theAlgo_.setVetoThreshold(aieta, vetoThreshold);
      }
      if (!overrideDBweightsAndFilterHB_) {
        theAlgo_.setNumFilterPresamplesHBQIE11(presamples);
        theAlgo_.setWeightQIE11(aieta, fixedPointWeight);
      }
    } else if (aieta < lastHERing) {
      if (!overrideDBvetoThresholdsHE_) {
        theAlgo_.setVetoThreshold(aieta, vetoThreshold);
      }
      if (!overrideDBweightsAndFilterHE_) {
        theAlgo_.setNumFilterPresamplesHEQIE11(presamples);
        theAlgo_.setWeightQIE11(aieta, fixedPointWeight);
      }
    }
  }
```